Code/Mantid/Framework/PythonInterface/plugins/algorithms/DNSFlippingRatioCorr.py

```diff
@@ -0,0 +1,347 @@
+            lhs_title = lhs_run.getProperty('run_title').value
+        if rhs_run.hasProperty('run_title'):
+            rhs_title = rhs_run.getProperty('run_title').value
+
+        for property_name in self.properties_to_compare:
+            if lhs_run.hasProperty(property_name) and rhs_run.hasProperty(property_name):
+                lhs_property = lhs_run.getProperty(property_name)
+                rhs_property = rhs_run.getProperty(property_name)
+                if lhs_property.type == rhs_property.type:
+                    if lhs_property.type == 'string':
+                        if lhs_property.value != rhs_property.value:
+                            message = "Property " + property_name + " does not match! " + \
+                                lhs_title + ": " + lhs_property.value + ", but " + \
+                                rhs_title + ": " + rhs_property.value
+                            self.log().warning(message)
+                    if lhs_property.type == 'number':
+                        if abs(lhs_property.value - rhs_property.value) > 5e-3:
+                            message = "Property " + property_name + " does not match! " + \
+                                lhs_title + ": " + str(lhs_property.value) + ", but " + \
+                                rhs_title + ": " + str(rhs_property.value)
+                            self.log().warning(message)
+                else:
+                    message = "Property " + property_name + " does not match! " + \
+                        lhs_title + ": " + str(lhs_property.value) + ", but " + \
+                        rhs_title + ": " + str(rhs_property.value)
+                    self.log().warning(message)
+            else:
+                message = "Property " + property_name + " is not present in " +\
+                    lhs_title + " or " + rhs_title + " - skipping comparison."
+                self.log().warning(message)
+        return True
+
+    def PyExec(self):
+        # Input
+        self.input_workspaces['SF_Data'] = self.getPropertyValue("SFDataWorkspace")
+        self.input_workspaces['NSF_Data'] = self.getPropertyValue("NSFDataWorkspace")
+        self.input_workspaces['SF_NiCr'] = self.getPropertyValue("SFNiCrWorkspace")
+        self.input_workspaces['NSF_NiCr'] = self.getPropertyValue("NSFNiCrWorkspace")
+        self.input_workspaces['SF_Background'] = self.getPropertyValue("SFBkgrWorkspace")
+        self.input_workspaces['NSF_Background'] = self.getPropertyValue("NSFBkgrWorkspace")
+        self.sf_outws_name = self.getPropertyValue("SFOutputWorkspace")
+        self.nsf_outws_name = self.getPropertyValue("NSFOutputWorkspace")
+        self.dfr = float(self.getPropertyValue("DoubleSpinFlipScatteringProbability"))
+
+        # check if possible to apply correction
+        if not self._can_correct():
+            raise RuntimeError("Error: impossible to apply flipping ratio correction to given workspaces!")
```

Comment:
This mixes return-value error handling with exceptions. An alternative, if you want to use exceptions more widely, would be to raise exceptions from inside `_can_correct()` instead of returning `False` from there and then raising `RuntimeError` here if `_can_correct()` returned false.
The main advantage in principle would be that for `_can_correct()` it would be easier to provide a more specific error string for each failure (must have same dimensions but I found XXX and YYY, must have same polarization, etc.). You can propagate the same idea to all error-checking functions, whether they are nested or not, like for example `_flipper_valid()`.